src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskController.java

```diff
@@ -225,31 +233,33 @@ public TaskController(CamundaService camundaService,
         ),
         @ApiResponse(
             code = 400,
-            message = "Bad Request"
+            message = BAD_REQUEST
         ),
         @ApiResponse(
             code = 403,
-            message = "Forbidden"
+            message = FORBIDDEN
         ),
         @ApiResponse(
             code = 415,
-            message = "Unsupported Media Type"
+            message = UNSUPPORTED_MEDIA_TYPE
         ),
         @ApiResponse(
             code = 500,
-            message = "Internal Server Error"
+            message = INTERNAL_SERVER_ERROR
         )
     })
     @PostMapping(path = "/{task-id}/assign")
     public ResponseEntity<String> assignTask(@RequestHeader("Authorization") String authToken,
-                                             @PathVariable("task-id") String taskId) {
+                                             @PathVariable(TASK_ID) String taskId,
+                                             @RequestBody AssigneeRequest assigneeRequest) {
+        boolean permissionCheckResult = permissionCheckService.validate(authToken, taskId, assigneeRequest.getUserId());
 
-        String userId = idamService.getUserId(authToken);
-        camundaService.assignTask(taskId, userId);
-        return ResponseEntity
-            .noContent()
-            .cacheControl(CacheControl.noCache())
-            .build();
+        if (permissionCheckResult) {
+            camundaService.assignTask(taskId, assigneeRequest.getUserId());
+            return ResponseEntity.noContent().cacheControl(CacheControl.noCache()).build();
+        } else {
```

Comment:
Remove else part, just return

Reply:
Well spotted! IntelliJ usually highlights these improvements. Not sure why it did not for this case. In any case, thanks!